backend/src/main/java/com/elshop/backend/category/CategoryController.java

```diff
@@ -22,6 +21,24 @@ public Category addNewCategory(@Valid @RequestBody CategoryRequest requestData)
         return categoryService.add(requestData);
     }
 
-    ;
+    @GetMapping
+    public List<Category> addNewCategory() {
+        return categoryService.getAll();
+    }
+
+    @GetMapping("{id}")
+    public Category addNewCategory(@PathVariable String id) {
+        return categoryService.getById(id);
+    }
 
+    @PutMapping("{id}")
+    public Category updateById(@PathVariable String id, @RequestBody @Valid CategoryRequest requestData) {
+        return categoryService.updateCategory(id, requestData);
```

Comment:
Consider adding a different http-entity as return in case a new record was created instead of updated

Reply:
I make check in my service ,  if the id exist I return 409, I only allow create throw post.